### DISCUSSION TITLE
Split the answer into multiple blocks

### DISCUSSION THREAD

**Discussion by macin** [rpggio/obsidian-chat-stream#18] [Ideas]
I think it would be cool if I could easly split the answer from chatgpt into multiple blocks. That way I could have a discussion and continue with just a part, e.g.

Me: List 3 places to go for a weekend
ChatGPT: Sure here are the best places to visit:
New York
Paris
London

<Now I split this into four cards and continue on selected path>
Me <after splitting the answer and continuing on London>: Write me an itinerary for saturday and sunday starting from 9am and finishing at 6pm. 
ChatGPT: Sure here is an itinerary for london (...)

**Comment by rpggio** [comment]
Yes, this is something I've considered. In the example you provided, you would have a note, 'Write me an itinerary for saturday and sunday...'. How would you link that in to the conversation branches for NY, Paris, London? Would you duplicate it as a child below each city and run AI on each?

**Reply by macin** [reply]
This is actually quite tricky. I will write in details how this could work over the weekend...

**Reply by macin** [reply]
Hi @rpggio , 
As promised, I recorded exactly the workflow I can do right now which is extremely manual and clunky. I was hoping, chat stream will make it way quicker and I think you are on a very good direction to make it so...

https://www.youtube.com/watch?v=xSMXLu69b6c

For the recording I was using Obsidian Copilot with Chatgpt integration, as I was not able to do similar workflow wich chatnote... maybe I am missing something.

When you watch the video you will see that basically there are multiple operations I had to do:

1. **Generate new child cards**, but for the future, I only want to keep those I like. It might be hard to auto generate cards becasue of chatgpt returining unstructured text, so I assume the user would need to do it. Therefore, the answer to your question is that I would keep the original response and create  sibbling card  based on partial selection of response card with full content
2. **Generate content on "on the same level" based on provided sibling cards**
3. **Rewrite selected card based on the prompt selected**

Of course copy paste is achievable, but if that was possible to create cards purely based on selected text that would be awesome. 

I hope above is nice and clear explanation of how I would like the workflow to work. I think, if it worked like that, this would be a killer feature. If you have any questions please let me know.
All the best
Marcin 

P.S. there are few conceptual corner cuttings on the video, as I had not enough time to make it perfect, but I hope what I recorded is good enough to understand the concept.

**Reply by rpggio** [reply]
Thank you for the effort you put into showing your workflow @macin .
I will be thinking about this. GPT does have JSON mode available with some models, so that it always returns structured data. There is definitely something possible here, for moving between an outline document and canvas notes.

**Reply by macin** [reply]
Ok cool, happy to be a beta tester if you need any help :)

**Reply by macin** [reply]
Hi @rpggio 
Did you have a chance to experiment with what we discussed here? I'm just wondering if I can help in any way :)
